pygis/docs/e_extraction.md

```diff
@@ -0,0 +1,350 @@
+
+```{code-cell} ipython3
+# Clip data
+clip_counties = gpd.clip(counties, poly)
+
+# Display attribute table
+display(clip_counties)
+
+# Plot clip
+plot_df(result_name = "San Francisco Bay Area Counties\nClip", result_df = clip_counties, result_geom_type = "polygon", area = poly)
+```
+
+We can clip any vector type. Next, we will clip the wells point data to our created rectangle polygon.
+
+```{code-cell} ipython3
+# Clip data
+clip_wells = gpd.clip(wells, poly)
+
+# Display attribute table
+display(clip_wells)
+
+# Plot clip
+plot_df(result_name = "San Francisco Bay Area Wells\nClip", result_df = clip_wells, result_geom_type = "point", area = poly)
+```
+
+## Select by Attribute
+
+Selecting by attribute selects only the features in a dataset whose attribute values match the specified criteria. `geopandas` uses the indexing and selection methods in `pandas`, so data in a GeoDataFrame can be selected and queried in the same way a `pandas` dataframe can.
+
+Sources: [Indexing and Selecting Data, GeoPandas](https://geopandas.org/docs/user_guide/indexing.html); [Indexing and selecting data, pandas](https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html); [How do I select a subset of a DataFrame?, pandas](https://pandas.pydata.org/pandas-docs/stable/getting_started/intro_tutorials/03_subset_data.html)
+
+We will use use different criteria to subset the wells dataset.
+
+```{code-cell} ipython3
+# Display attribute table
+display(wells)
+```
+
+The criteria can use a variety of operators, including comparison and logical operators.
+
+```{code-cell} ipython3
+# Select wells that are public supply
+wells_subset_1 = wells[(wells["WELL_USE"] == "Public Supply")]
+
+# Display first five and last five rows of attribute table
+display_table(table_name = "San Francisco Bay Area Wells - Public Supply", attribute_table = wells_subset_1)
+```
+
+```{code-cell} ipython3
+# Select wells that are public supply and have a depth greater than 50 ft
+wells_subset_2 = wells[(wells["WELL_USE"] == "Public Supply") & (wells["WELL_DEPTH"] > 50)]
+
+# Display first five and last five rows of attribute table
+display_table(table_name = "San Francisco Bay Area Wells - Public Supply with Depth Greater than 50 ft", attribute_table = wells_subset_2)
+```
+
+```{code-cell} ipython3
+# Select wells that are public supply and have a depth greater than 50 ft OR are residential
+wells_subset_3 = wells[((wells["WELL_USE"] == "Public Supply") & (wells["WELL_DEPTH"] > 50)) | (wells["WELL_USE"] == "Residential")]
+
+# Display first five and last five rows of attribute table
+display_table(table_name = "San Francisco Bay Area Wells - Public Supply with Depth Greater than 50 ft or Residential", attribute_table = wells_subset_3)
+```
+
+## Select by Location
+
+Selecting by location selects features based on its relative spatial relationship with another dataset.
+
+There are multiple spatial relationships available in `geopandas`.
```

Comment:
Describe using an example. Need to elaborate here a bit

[also link to arcmap description with has some drawings](https://desktop.arcgis.com/en/arcmap/10.3/map/working-with-layers/using-select-by-location.htm)